### ISSUE TITLE
Tests fail to run if they have a name that is more than 77 character in length.

### ISSUE THREAD

**Issue by diseraluca** [JohnnyHendriks/TestAdapter_Catch2#17]
First of all thanks a lot for your amazing work on this adapter, it really streamlines the testing process for catch in visual studio.

I've found that any test that has a name that is more than 77 characters in length will fail to run, by being  skipped, with the following motivation:

![capture](https://user-images.githubusercontent.com/39652885/50550932-269e5e80-0c79-11e9-8ab6-c0bdbbc533db.PNG)

Now, as you can see, in the Test Explorer the name gets mangled, and I suspect this is the real reason for which it cannot be run. I'm not sure if this is a Visual Studio limitation or it has something to do with the Catch2 Adapter, but it seemed reasonable to report it.
I tested on a Python Project with python's unittest and I could have names longer than 77 characters without incurring in issues but I can't exclude that this may still be a Visual Studio Issue. 

If I run the catch executable manually the test gets correctly run and if I shorten the name to <= 77 characters the same is true.

To Replicate:

1. Create a Visual Studio solution with catch tests
2. Create a TEST_CASE with a name of length > 77 characters
3. Build the solution to discover the test cases
4. Run the TEST_CASE with a name longer than 77 characters

If it happens to be a Visual Studio limitation please forgive me for reporting a non-issue.

**Comment by JohnnyHendriks** [comment]
I'm glad to hear you find the test adapter useful. I tried to reproduce your problem with the following test name:
`Testset02.Tests02. LongName 0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789`
, which has 128 characters. This seemed to work just fine. Then I remembered I was using [custom test discovery](https://github.com/JohnnyHendriks/TestAdapter_Catch2/blob/master/Docs/Settings.md#discovercommandline), so I tried with the generic test discovery settings. There I was able to reproduce the problem.

I will try to fix it, but there is some loss of information that would make it impossible to fix it for all situations. Basically what happens is that when generic test case discovery is used, long test case names are split over multiple lines, here line breaks are attempted to be placed at locations of a space character. That space character is however not retained. Should there be more than 1 space character at the break location all those space characters are lost and the correct test case name cannot be reconstructed. Furthermore, a break can also happen at a '-' character, which is retained. However, if a break is forced the same character is used to indicate the break. I was already able to create multiple test case names that result in the same test case name discovery output.

Thank you for reporting this.
